bentoml/_internal/configuration/containers.py

```diff
@@ -408,6 +409,15 @@ def tracer_provider(
         else:
             return provider
 
+    @providers.SingletonFactory
+    @staticmethod
+    def tracing_excluded_urls(
+        excluded_urls: t.Optional[str] = Provide[config.tracing.excluded_urls],
+    ):
+        from opentelemetry.util.http import parse_excluded_urls
+
+        return parse_excluded_urls(excluded_urls)
```

Comment:
Does this function accept `None` type?

Reply:
Yes it does, based on what I understand from the [source code](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/fbac4245159e51877efa86581724995bd1415a18/util/opentelemetry-util-http/src/opentelemetry/util/http/__init__.py#L69), it should return empty list if you pass in `None`.

Reply:
noted.

Just take a quick peek at the source code. Seems like 

https://github.com/open-telemetry/opentelemetry-python-contrib/blob/fbac4245159e51877efa86581724995bd1415a18/util/opentelemetry-util-http/src/opentelemetry/util/http/__init__.py#L58 also exclude additional URLs for instrumentation.

We do use some instrumentation library, and do you think it is necessary that BentoML also filter that out?

Reply:
Based on what I understand, BentoML uses `opentelemetry.instrumentation.asgi`, which does not call `get_excluded_urls` ([source](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/fbac4245159e51877efa86581724995bd1415a18/instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py)). This is unlike `opentelemetry.instrumentation.fastapi`, which does call `get_excluded_urls` ([source](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/fbac4245159e51877efa86581724995bd1415a18/instrumentation/opentelemetry-instrumentation-fastapi/src/opentelemetry/instrumentation/fastapi/__init__.py)), and hence the user can just inject the environment variable and it will work.

So unfortunately, there is no environment variable that a user can inject to make use of the builtin exclude urls logic in the instrumentation.

Reply:
I think configuration-driven solution from you is fantastic. I do think there are also value for injecting an environment variables.

I'm wondering since BentoML does allow users to mount FastAPI apps, should we also respect the environment variable set by `opentelemetry.instrumentation.fastapi` 🤔 ?

Nonetheless, I think for first pass this is more than enough.

Reply:
Good point - but in the event that both the environment variable and the bentoml config provide excluded urls, how should we handle it? Do we:

- Merge the 2 lists, or
- Have the bentoml config override the environment variable, or
- Have the environment variable override the bentoml config?

Reply:
I think the order will always be respect the environment variable first, then config.